### ISSUE TITLE
Cluster creation via CLI fails - AAD role propogation

### ISSUE THREAD

**Issue by jim-h-wang** [Azure/AKS#129]
# Issue 1
n fact, the process just didn't wait long enough.  I re-ran after checking App Registration was visible in the portal using details in acsServicePrincipal.json and the cluster was created fine.

```
az aks create --resource-group testResourceGroup1 --name testk8sCluster1 --node-count 1 --generate-ssh-keys
AAD role propagation done[############################################]  100.0000%Operation failed with status: 'Bad Request'. Details: Service principal clientID: 1753d70a-e218-4fed-846b-59a32b42cb46 not found in Active Directory tenant 990e1376-cb1c-406d-8556-32e5a40c0520, Please see https://aka.ms/acs-sp-help for more details.
```

**Comment by mboersma** [comment]
This is a known issue where the CLI doesn't always wait long enough for the AAD role to have propagated everywhere. The workaround (as you discovered) is to try again after waiting a bit.

The good news is that this particular role assignment was brought over from `az acs` and isn't strictly necessary. Azure/azure-cli#5208 removes this code from `az aks create` and thus avoids the problem you saw. This should be available in the next `az` release this week.

**Comment by mboersma** [comment]
This is fixed in the current `az` CLI 2.0.26.
